packages/cli/src/commands/gsn-start.ts

```diff
@@ -1,19 +1,35 @@
 import commander from 'commander'
 import { gsnCommander, saveDeployment, showDeployment } from '../utils'
 import { GsnTestEnvironment } from '../GsnTestEnvironment'
+import { execSync } from 'child_process'
 
 gsnCommander(['n'])
   .option('-w, --workdir <directory>', 'relative work directory (defaults to build/gsn/)', 'build/gsn')
+  .option('--withNode', 'start with "hardhat node" in the background', false)
+  .option('--run <command>', 'run with command after GSN is started. stop GSN when the commands finish')
   .parse(process.argv);
 
 (async () => {
   try {
     const network: string = commander.network
-    const env = await GsnTestEnvironment.startGsn(network)
+    if (network !== 'localhost' && commander.withNode != null) {
+      throw new Error('can\'t have both --network and --withNode')
+    }
+    const env = await GsnTestEnvironment.startGsn(network, commander.withNode)
     saveDeployment(env.contractsDeployment, commander.workdir)
-    showDeployment(env.contractsDeployment, 'GSN started')
 
-    console.log(`Relay is active, URL = ${env.relayUrl} . Press Ctrl-C to abort`)
+    if (commander.run != null) {
+      console.log('running command: ', commander.run)
+      try {
+        execSync(commander.run, { stdio: 'inherit' })
+        process.exit(0)
+      } catch (e: any) {
+        process.exit(1)
```

Comment:
won't this consume the exception instead of printing a stack trace?

Reply:
yes, it does. but the exception is "execSync failed", or something like that. 
The external command itself inherits both stdio and stdin, so you'd get as a a response something like this
`/bin/sh: asdasd: command not found`
or whatever error the command emits.
If the command you're running is completely silent, then yes, it would not display any error (e.g. `run-with-gsn "exit 1"` just dumps the "gsn start" output, and exits with error code 1)